webapp/src/index.tsx

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import * as React from 'react';
+
+import {Channel} from 'mattermost-redux/types/channels';
+
+import Icon from './components/icon';
+import PostTypeJitsi from './components/post_type_jitsi';
+import RootPortal from './components/root_portal';
+import reducer from './reducers';
+import {startMeeting, loadConfig} from './actions';
+
+class PluginClass {
+    rootPortal: RootPortal | null = null
+
+    initialize(registry: any, store: any) {
+        if ((window as any).JitsiMeetExternalAPI) {
+            this.rootPortal = new RootPortal(registry, store);
+            if (this.rootPortal) {
+                this.rootPortal.render();
+            }
+        } else {
+            const script = document.createElement('script');
+            script.type = 'text/javascript';
+            script.onload = () => {
+                this.rootPortal = new RootPortal(registry, store);
+                if (this.rootPortal) {
+                    this.rootPortal.render();
+                }
+            };
+            script.src = (window as any).basename + '/plugins/jitsi/jitsi_meet_external_api.js';
```

Comment:
Does this work on mattermost instanceson subdirectories? (i.e. `www.myinstance.com/mymattermost` instead of `www.mymattermost.com`)

Reply:
Also, were the plugin ID change for any reason (for example to something like `com.mattermost.jitsi`) this would have to be changed. I would use here directly the plugin ID.

Reply:
Yes, it works on subdirectories, actually the `basename` was a fix for exactly that.

And yes, we can change there the text to use the jitsi plugin id instead of the string directly.